### ISSUE TITLE
Error with usage of send_file in flask

### ISSUE THREAD

**Issue by gabriel2029** [benoitc/gunicorn#2197]
Hello,
in gunicorn version 20.0.0, I use the flask method `send_file` as following:
````
media_file = MediaFile.query.filter_by(exam_id=exam_id, file_name=file_name).first()
if not media_file: abort(404)

return send_file(media_file.get_file_path())
````

When I open the page on the browser, the server outputs an error with the following message:
````
[2019-11-23 01:16:49 +0100] [8025] [ERROR] Error handling request
Traceback (most recent call last):
  File "/opt/chiefexam/env/lib/python3.7/site-packages/gunicorn/workers/base_async.py", line 111, in handle_request
    resp.write_file(respiter)
  File "/opt/chiefexam/env/lib/python3.7/site-packages/gunicorn/http/wsgi.py", line 397, in write_file
    if not self.sendfile(respiter):
  File "/opt/chiefexam/env/lib/python3.7/site-packages/gunicorn/http/wsgi.py", line 387, in sendfile
    sent += os.sendfile(sockno, fileno, offset + sent, count)
  File "/opt/chiefexam/env/lib/python3.7/site-packages/gunicorn/workers/geventlet.py", line 32, in _eventlet_sendfile
    return os.sendfile(fdout, fdin, offset, nbytes)
  File "/opt/chiefexam/env/lib/python3.7/site-packages/gunicorn/workers/geventlet.py", line 32, in _eventlet_sendfile
    return os.sendfile(fdout, fdin, offset, nbytes)
  File "/opt/chiefexam/env/lib/python3.7/site-packages/gunicorn/workers/geventlet.py", line 32, in _eventlet_sendfile
    return os.sendfile(fdout, fdin, offset, nbytes)
  [Previous line repeated 968 more times]
RecursionError: maximum recursion depth exceeded
````

After I have downgraded to gunicorn 19.9.0, the error disappears. I think this is a bug with gunicorn, or did i miss something?

**Comment by benoitc** [comment]
this should be fixed in 902d9c89abb2cf9bed688590b2276a19fcf4d35a . Can you test latest master? 20.0.1 is coming.

**Comment by benoitc** [comment]
20.0.1 is out fixing the issue.
